sequencer/src/catchup.rs

```diff
@@ -0,0 +1,134 @@
+use crate::{
+    api::endpoints::{AccountQueryData, BlocksFrontier},
+    state::{BlockMerkleTree, FeeAccount, FeeMerkleCommitment},
+    Header,
+};
+use commit::Commitment;
+use futures::future::join_all;
+use hotshot_types::{data::ViewNumber, traits::node_implementation::ConsensusTime as _};
+use jf_primitives::merkle_tree::{ForgetableMerkleTreeScheme, MerkleTreeScheme as _};
+use serde::de::DeserializeOwned;
+use std::time::Duration;
+use surf_disco::Request;
+use tide_disco::error::ServerError;
+use url::Url;
+
+#[derive(Debug, Clone)]
+struct Client<ServerError> {
+    inner: surf_disco::Client<ServerError>,
+    url: Url,
+}
+
+impl Client<ServerError> {
+    pub fn new(url: Url) -> Self {
+        Self {
+            inner: surf_disco::Client::new(url.clone()),
+            url,
+        }
+    }
+
+    pub fn get<T: DeserializeOwned>(&self, route: &str) -> Request<T, ServerError> {
+        self.inner.get(route)
+    }
+}
+
+#[derive(Debug, Clone, Default)]
+pub struct StatePeers {
+    clients: Vec<Client<ServerError>>,
+    interval: Duration,
+}
+
+impl StatePeers {
+    pub fn from_urls(urls: Vec<Url>) -> Self {
+        Self {
+            clients: urls.into_iter().map(Client::new).collect(),
+            interval: Duration::from_secs(1),
+        }
+    }
+
+    pub async fn fetch_accounts(
+        &self,
+        view: ViewNumber,
+        fee_merkle_tree_root: FeeMerkleCommitment,
+        accounts: impl IntoIterator<Item = FeeAccount>,
+    ) -> Vec<AccountQueryData> {
+        join_all(
```

Comment:
We have had issues in Gibraltar with the commit task firing a bunch of simultaneous requests to the query service, causing spikes in CPU usage on the commit task and the server. I don't expect we'll have too many fee accounts to fetch at one time here, but still maybe it is safer to fetch these one at a time or at least break them into fixed-size batches to fetch in parallel

Reply:
Actually maybe we should add a batch endpoint for fetching multiple paths at once.

Reply:
Ok. I think at this stage doing them sequentially is probably best, we can optimize later.